client/state/analytics/middleware.js

```diff
@@ -0,0 +1,40 @@
+import analytics from 'analytics';
+import invoke from 'lodash/invoke';
+import reduxAnalytics from 'redux-analytics';
+
+import {
+	BUMP_STAT,
+	RECORD_EVENT,
+	RECORD_PAGE_VIEW
+} from './constants';
+
+const eventServices = {
+	ga: ( { category, action, label, value } ) => analytics.ga.recordEvent( category, action, label, value ),
+	tracks: ( { name, properties } ) => analytics.tracks.recordEvent( name, properties )
+};
+
+const pageViewServices = {
+	ga: ( { url, title } ) => analytics.ga.recordPageView( url, title ),
+	default: ( { url, title } ) => analytics.pageView.record( url, title )
+};
+
+const statBump = ( { group, name } ) => analytics.mc.bumpStat( group, name );
+
+export const dispatcher = ( { analytics: { type, payload } } ) => {
+	const { service = 'default' } = payload;
+
+	switch ( type ) {
+		case RECORD_EVENT:
+			return invoke( eventServices, service, payload );
+
+		case RECORD_PAGE_VIEW:
+			return invoke( pageViewServices, service, payload );
+
+		case BUMP_STAT:
+			return statBump( payload );
+	}
+};
+
+export const analyticsMiddleware = reduxAnalytics( dispatcher );
```

Comment:
@aduth I considered this and started out from scratch, but we had already incorporated `redux-analytics` in the theme state middleware. I decided to go ahead and stick with it after a conversation with @ockham. Also see the brief discussion with @artpi [above](https://github.com/Automattic/wp-calypso/pull/4699#discussion_r59552030)